File: 0169. Majority Element/Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        res, cnt = 0, 0;
        
        for num in nums:
            if cnt == 0:
                res, cnt = num, cnt + 1
            elif res == num:
                cnt += 1
            else:
                cnt -= 1
            
        return res

File: 0169. Majority Element/Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int res, cnt = 0;
        
        for (auto num : nums) {
            if (cnt == 0) cnt++, res = num;
            else if (res == num) cnt++;
            else cnt--;
        }
        
        return res;
    }
};